In [3]:
#Import Pandas
import pandas as pd

In [4]:
# Create the dataframe
data = pd.read_csv('shopify_data.csv') 

In [5]:
# Create the statistic demonstrated in the question
AOV = data['order_amount'].mean()
print(AOV)

3145.128


##### AOV as previously calculated is accurate, 3145.13 is the average order amount

In [6]:
# Show the highest order amounts
data.sort_values('order_amount', ascending = False).head(20)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
2153      2154       42      607        704000         2000    credit_card   
3332      3333       42      607        704000         2000    credit_card   
520        521       42      607        704000         2000    credit_card   
1602      1603       42      607        704000         2000    credit_card   
60          61       42      607        704000         2000    credit_card   
2835      2836       42      607        704000         2000    credit_card   
4646      4647       42      607        704000         2000    credit_card   
2297      2298       42      607        704000         2000    credit_card   
1436      1437       42      607        704000         2000    credit_card   
4882      4883       42      607        704000         2000    credit_card   
4056      4057       42      607        704000         2000    credit_card   
15          16       42      607        704000         2000    credit_card   
1104      1105       42      607        704000         2000    credit_card   
1562      1563       42      607        704000         2000    credit_card   
2969      2970       42      607        704000         2000    credit_card   
4868      4869       42      607        704000         2000    credit_card   
1362      1363       42      607        704000         2000    credit_card   
691        692       78      878        154350            6          debit   
2492      2493       78      834        102900            4          debit   
3724      3725       78      766         77175            3    credit_card   

               created_at  
2153   2017-03-12 4:00:00  
3332   2017-03-24 4:00:00  
520    2017-03-02 4:00:00  
1602   2017-03-17 4:00:00  
60     2017-03-04 4:00:00  
2835   2017-03-28 4:00:00  
4646   2017-03-02 4:00:00  
2297   2017-03-07 4:00:00  
1436   2017-03-11 4:00:00  
4882   2017-03-25 4:00:00  
4056   2017-03-28 4:00:00  
15     2017-03-07 4:00:00  
1104   2017-03-24 4:00:00  
1562   2017-03-19 4:00:00  
2969   2017-03-28 4:00:00  
4868   2017-03-22 4:00:00  
1362   2017-03-15 4:00:00  
691   2017-03-27 22:51:43  
2492   2017-03-04 4:37:34  
3724  2017-03-16 14:13:26

In [17]:
# The sale between user_id 607 and shop_id 42 appears to be an extreme outlier, likely a business to business sale. 
# Shop_id 78 has something quite strange surrounding its items for sale. Sneakers generally do not cost tens of thousands of dollars. 
# Further analysis needed for both oddities. 


# Create a new column representing the average item cost
data['avg_item_cost'] = data['order_amount'] / data['total_items']
data.head(5)

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

            created_at  avg_item_cost  
0  2017-03-13 12:36:56          112.0  
1  2017-03-03 17:38:52           90.0  
2   2017-03-14 4:23:56          144.0  
3  2017-03-26 12:43:37          156.0  
4   2017-03-01 4:35:11          156.0

In [20]:
# Explore if shop 78 sells any items other than its strange 25725$ item. 
shop_78 = data[data['shop_id'] == 78]
shop_78

order_id  shop_id  user_id  order_amount  total_items payment_method  \
160        161       78      990         25725            1    credit_card   
490        491       78      936         51450            2          debit   
493        494       78      983         51450            2           cash   
511        512       78      967         51450            2           cash   
617        618       78      760         51450            2           cash   
691        692       78      878        154350            6          debit   
1056      1057       78      800         25725            1          debit   
1193      1194       78      944         25725            1          debit   
1204      1205       78      970         25725            1    credit_card   
1259      1260       78      775         77175            3    credit_card   
1384      1385       78      867         25725            1           cash   
1419      1420       78      912         25725            1           cash   
1452      1453       78      812         25725            1    credit_card   
1529      1530       78      810         51450            2           cash   
2270      2271       78      855         25725            1    credit_card   
2452      2453       78      709         51450            2           cash   
2492      2493       78      834        102900            4          debit   
2495      2496       78      707         51450            2           cash   
2512      2513       78      935         51450            2          debit   
2548      2549       78      861         25725            1           cash   
2564      2565       78      915         77175            3          debit   
2690      2691       78      962         77175            3          debit   
2773      2774       78      890         25725            1           cash   
2818      2819       78      869         51450            2          debit   
2821      2822       78      814         51450            2           cash   
2906      2907       78      817         77175            3          debit   
2922      2923       78      740         25725            1          debit   
3085      3086       78      910         25725            1           cash   
3101      3102       78      855         51450            2    credit_card   
3151      3152       78      745         25725            1    credit_card   
3167      3168       78      927         51450            2           cash   
3403      3404       78      928         77175            3          debit   
3440      3441       78      982         25725            1          debit   
3705      3706       78      828         51450            2    credit_card   
3724      3725       78      766         77175            3    credit_card   
3780      3781       78      889         25725            1           cash   
4040      4041       78      852         25725            1           cash   
4079      4080       78      946         51450            2           cash   
4192      4193       78      787         77175            3    credit_card   
4311      4312       78      960         51450            2          debit   
4412      4413       78      756         51450            2          debit   
4420      4421       78      969         77175            3          debit   
4505      4506       78      866         25725            1          debit   
4584      4585       78      997         25725            1           cash   
4715      4716       78      818         77175            3          debit   
4918      4919       78      823         25725            1           cash   

               created_at  avg_item_cost  
160    2017-03-12 5:56:57        25725.0  
490   2017-03-26 17:08:19        25725.0  
493   2017-03-16 21:39:35        25725.0  
511    2017-03-09 7:23:14        25725.0  
617   2017-03-18 11:18:42        25725.0  
691   2017-03-27 22:51:43        25725.0  
1056  2017-03-15 10:16:45        25725

In [19]:
# Explore if user_id 607 purchases anything other than its immensely large order. 
user_607 = data[data['user_id'] == 607]
user_607

order_id  shop_id  user_id  order_amount  total_items payment_method  \
15          16       42      607        704000         2000    credit_card   
60          61       42      607        704000         2000    credit_card   
520        521       42      607        704000         2000    credit_card   
1104      1105       42      607        704000         2000    credit_card   
1362      1363       42      607        704000         2000    credit_card   
1436      1437       42      607        704000         2000    credit_card   
1562      1563       42      607        704000         2000    credit_card   
1602      1603       42      607        704000         2000    credit_card   
2153      2154       42      607        704000         2000    credit_card   
2297      2298       42      607        704000         2000    credit_card   
2835      2836       42      607        704000         2000    credit_card   
2969      2970       42      607        704000         2000    credit_card   
3332      3333       42      607        704000         2000    credit_card   
4056      4057       42      607        704000         2000    credit_card   
4646      4647       42      607        704000         2000    credit_card   
4868      4869       42      607        704000         2000    credit_card   
4882      4883       42      607        704000         2000    credit_card   

              created_at  avg_item_cost  
15    2017-03-07 4:00:00          352.0  
60    2017-03-04 4:00:00          352.0  
520   2017-03-02 4:00:00          352.0  
1104  2017-03-24 4:00:00          352.0  
1362  2017-03-15 4:00:00          352.0  
1436  2017-03-11 4:00:00          352.0  
1562  2017-03-19 4:00:00          352.0  
1602  2017-03-17 4:00:00          352.0  
2153  2017-03-12 4:00:00          352.0  
2297  2017-03-07 4:00:00          352.0  
2835  2017-03-28 4:00:00          352.0  
2969  2017-03-28 4:00:00          352.0  
3332  2017-03-24 4:00:00          352.0  
4056  2017-03-28 4:00:00          352.0  
4646  2017-03-02 4:00:00          352.0  
4868  2017-03-22 4:00:00          352.0  
4882  2017-03-25 4:00:00          352.0

##### The seemingly strange number of 3145.13 results from two peculiar outliers. One outlier is a customer who only made repeated identical extremely large orders; 17 orders at 704,000 each. Likely a business the shop is familiar with on a personal basis. The other outlier results from an item that sells for 25725.00, which could be a type of ultra exclusive shoe, for instance the Nike MAG 2016, or much more likely is something other than a shoe. Either way the shop would be aware of the circumstances of these expensive sales, all the more so since some of the sales are in cash. Over 25 000 in cash is an extremely strange amount of cash under most circumstances. 
##### In both cases, the each outlier belongs to a singular shop_id each, so the shops themselves are likely outliers (meaning, not that relavent for calculating AOV for sneakers across sneaker shops within 3 standard deviations, or in this case even 10 deviations).


##### A better metric would be all orders other than the orders hailing from these two shops. 

In [8]:
# Create a new dataframe bereft of the outliers
aov_modded = data.loc[(data['order_amount'] < 700000) & (data['avg_item_cost'] < 25000)]

In [9]:
# Calculate the new average order value
aov_modded['order_amount'].mean()

302.58051448247926

In [23]:
# Purely for incedental interest, explore what the average item cost is. 
aov_modded['avg_item_cost'].mean()

151.7885355479036

##### Thus, a more accurate AOV is 302.58$. 

### Methodology
##### Especially for such small datasets, the Pandas library in Python is my go to tool. On such small datasets, if exploratory analysis needs to be done, Pandas is usually faster to survey with and is more pervasive as a toolset. 
##### Discovering where the error came from was immediate once the highest orders were brought to light. Whereas the rest of the orders come from many sources as one would expect from sneaker shops, and whom fall in a relatively tightly knit standard distribution for order value, the outliers are so utterly far above this bell curve and belong to only two sources that they are extremely likely to be due to special circumstances instead of being extreme ends of any standard distribution. 
##### Taking out these two outliers outputted an AOV that represents what sneaker shop owners could expect as an average order, which is the statistic I imagine they're looking for. 

##### As AOV is a metric the shop owners are familiar with and is industry standard, I'd keep the metric as AOV, but leave out the two extreme outliers. If these two outliers need to be classified for the future of standard practice, the user_id 607 is likely a business; meaning it's business to business sales. And shop_78 either sells something expensive, in addition to shoes, or sells ultra rare shoes. In both cases these sales should be classified as something other than sneakers, by the shop owners. Shop_78 is also not benefitting from the initial AOV calculation, as it deals solely in items no other shop sells and thus the other shop's data doesn't collate to anything meaningful for it. Likely shop_78 would be motivated to classify this 25725 item as something other than sneakers, for its own metrics.